eclipsesource/cdtcloud-deploymentserver
# feat(DeviceConnector/Device): Devices, DeviceTypes & Monitoring Part 1

## Changes Proposed:


- Multiple type additions
- Various functions for Device and DeviceTypes
- Proper handling of not existing or unavailable devices
- DeviceConnector now connects to DeploymentServer via a Duplex stream instead of a WebSocket
- Various changes for future monitor handling & piping
- Device-Status is now updated locally and remotely
- Code-CleanUp, Refactoring and Type-Changes

## How to Test the Changes:


1. StartUp processes as per usual
2. Deploy and change Code
3. Result on the device should be the same as on older versions

## Known Issues and TODO List:


- Monitoring is not in its finished state
- Monitoring output is not yet piped to deployment-server
- Deployment-Status is not yet updated on the server accordingly

## How to Test PRs
1) First fetch the Pull-Request from the main repo, replacing `xx` with the ID of the PR:\
`git fetch git@github.com:eclipsesource/cdtcloud-deploymentserver.git pull/xx/head:pr_xx`
2) Checkout the Pull-Request, again replacing `xx` with the PR-ID:\
`git checkout pr_xx`
3) Perform the necessary tests for the PR
